In [67]:
import pandas as pd
import numpy as np
dane1 = pd.read_csv("dane.csv", header=None, na_values='?')

dane1.columns = ["pclass","survived","name","sex","age","sibsp","parch","ticket","fare","cabin","embarked","boat","body","home.dest"]
col = ["pclass","survived","name","sex","ticket","cabin","embarked","boat","body","home.dest"]

for i in col:
  print('Liczba etykiet zmiennej '+ i + ' {}'.format(len(dane1.loc[:,i].unique())))

print("\nliczba wszystkich imion" + ' {}'.format(len(dane1.loc[:,"name"])))

list_cab = dane1.loc[:,"cabin"].unique()

print("\nliczba unikalnych kabin" + ' {}'.format(len(list_cab)))

dane1["CabinReduced"] = dane1["cabin"].astype(str).str[0]

print("\nliczba unikalnych zredukowanych kabin" + ' {}'.format(len(dane1.loc[:,"CabinReduced"].unique())))

print("Zredukowano o " + str(round(100 - 9/187*100,2)) + "% kardynalnosc zbioru zmiennej opisujacej kabiny")
dane1.head(20)


Liczba etykiet zmiennej pclass 3
Liczba etykiet zmiennej survived 2
Liczba etykiet zmiennej name 1307
Liczba etykiet zmiennej sex 2
Liczba etykiet zmiennej ticket 929
Liczba etykiet zmiennej cabin 187
Liczba etykiet zmiennej embarked 4
Liczba etykiet zmiennej boat 28
Liczba etykiet zmiennej body 122
Liczba etykiet zmiennej home.dest 370

liczba wszystkich imion 1309

liczba unikalnych kabin 187

liczba unikalnych zredukowanych kabin 9
Zredukowano o 95.19% kardynalnosc zbioru zmiennej opisujacej kabiny


pclass  survived                                               name  \
0        1         1                      Allen, Miss. Elisabeth Walton   
1        1         1                     Allison, Master. Hudson Trevor   
2        1         0                       Allison, Miss. Helen Loraine   
3        1         0               Allison, Mr. Hudson Joshua Creighton   
4        1         0    Allison, Mrs. Hudson J C (Bessie Waldo Daniels)   
5        1         1                                Anderson, Mr. Harry   
6        1         1                  Andrews, Miss. Kornelia Theodosia   
7        1         0                             Andrews, Mr. Thomas Jr   
8        1         1      Appleton, Mrs. Edward Dale (Charlotte Lamson)   
9        1         0                            Artagaveytia, Mr. Ramon   
10       1         0                             Astor, Col. John Jacob   
11       1         1  Astor, Mrs. John Jacob (Madeleine Talmadge Force)   
12       1         1                      Aubart, Mme. Leontine Pauline   
13       1         1                       Barber, Miss. Ellen 'Nellie'   
14       1         1               Barkworth, Mr. Algernon Henry Wilson   
15       1         0                                Baumann, Mr. John D   
16       1         0                           Baxter, Mr. Quigg Edmond   
17       1         1    Baxter, Mrs. James (Helene DeLaudeniere Chaput)   
18       1         1                              Bazzani, Miss. Albina   
19       1         0                               Beattie, Mr. Thomson   

       sex      age  sibsp  parch    ticket      fare    cabin embarked boat  \
0   female  29.0000      0      0     24160  211.3375       B5        S    2   
1     male   0.9167      1      2    113781  151.5500  C22 C26        S   11   
2   female   2.0000      1      2    113781  151.5500  C22 C26        S  NaN   
3     male  30.0000      1      2    113781  151.5500  C22 C26        S  NaN   
4   female  25.0000      1      2    113781  151.5500  C22 C26        S  NaN   
5     male  48.0000      0      0     19952   26.5500      E12        S    3   
6   female  63.0000      1      0     13502   77.9583       D7        S   10   
7     male  39.0000      0      0    112050    0.0000      A36        S  NaN   
8   female  53.0000      2      0     11769   51.4792     C101        S    D   
9     male  71.0000      0      0  PC 17609   49.5042      NaN        C  NaN   
10    male  47.0000      1      0  PC 17757  227.5250  C62 C64        C  NaN   
11  female  18.0000      1      0  PC 17757  227.5250  C62 C64        C    4   
12  female  24.0000      0      0  PC 17477   69.3000      B35        C    9   
13  female  26.0000      0      0     19877   78.8500      NaN        S    6   
14    male  80.0000      0      0     27042   30.0000      A23        S    B   
15    male      NaN      0      0  PC 17318   25.9250      NaN        S  NaN   
16    male  24.0000      0      1  PC 17558  247.5208  B58 B60        C  NaN   
17  female  50.0000      0      1  PC 17558  247.5208  B58 B60        C    6   
18  female  32.0000      0      0     11813   76.2917      D15        C    8   
19    male  36.0000      0      0     13050   75.2417       C6        C    A   

     body                        home.dest CabinReduced  
0     NaN                     St Louis, MO            B  
1     NaN  Montreal, PQ / Chesterville, ON            C  
2     NaN  Montreal, PQ / Chesterville, ON            C  
3   135.0  Montreal, PQ / Chesterville, ON            C  
4     NaN  Montreal, PQ / Chesterville, ON            C  
5     NaN                     New York, NY            E  
6     NaN                       Hudson, NY            D  
7     NaN                      Belfast, NI            A  
8     NaN              Bayside, Queens, NY            C  
9    22.0              Montevideo, Uruguay            n  
10  124.0                     New York, NY            C  
11    NaN                     New York, NY         

Liczba unikalnych imion pasazerow rozni sie od liczby wszystkich pasazerow o 2

Duża moc zbioru:

name: 1307
ticket: 929
cabin: 187
body: 122
home.dest: 370

Mała moc zbioru:

pclass: 3
survived: 2
sex: 2
embarked: 4
boat: 28

Redukujemy akurat te zmienna poniewaz nie chcemy tak dokladnie analizowac tej cechy wystarczy nam wiedza w ktorej czesci statku znajdowala sie dana kabina do czego wystraczy nam pierwsza litera

Upraszcza to przyszle analizy

Moze powodowac pewne male niescislosci
